### PR TITLE
[4.0] Filter third party extensions

### PR DESCRIPTION
**Pull request by roland-d**
Pull Request for Issue #13795

This is a redo of 13795 because we can't filter on extension ID but we now have a function that lists all the core extensions. So using this for the filter.

### Summary of Changes
This change adds an extra filter on the Extensions -> Manage page to filter extensions on whether they are core or 3rd party extensions.
image

### Testing Instructions
1. Apply path
2. Go to Extensions -> Manage
3. Click on Table Options
4. There is now a filter on the end called **- Select Core -**
5. Filter on **Show core extensions**
6. The list now only shows core extensions
7. Filter on **Show third party extensions**
8. The list now only shows third party extensions
9. Filter on **- Select Core -**
10. All extensions show in the list

### Documentation Changes Required
None that I know of

### CODE REVIEW

**Comment by brianteeman** [comment]
Personally I prefer the term "non-core" to 3rd party. Plus any optional extension such as we links is non-core and not 3rd party

**Comment by roland-d** [comment]
Changing the wording to non-core isn't a problem. Will fix that.

**Comment by brianteeman** [comment]
thanks

File: administrator/language/en-GB/en-GB.com_installer.ini [hunk 1]
```diff
@@ -254,6 +254,9 @@ COM_INSTALLER_UPLOAD_AND_INSTALL="Upload &amp; Install"
 COM_INSTALLER_UPLOAD_INSTALL_JOOMLA_EXTENSION="Upload & Install Joomla Extension"
 COM_INSTALLER_UPLOAD_PACKAGE_FILE="Upload Package File"
 COM_INSTALLER_VALUE_CLIENT_SELECT="- Select Location -"
+COM_INSTALLER_VALUE_CORE_SELECT=" - Select Core - "
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_installer.ini]: Sort keys in alpha order.

Comment (roland-d) [reply administrator/language/en-GB/en-GB.com_installer.ini]: @Quy Did you see this whole file is a mess?

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on dd218b81894c2ae8e2adce7b682b9ca3442bf5be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17832">issues.joomla.org/tracker/joomla-cms/17832</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on dd218b81894c2ae8e2adce7b682b9ca3442bf5be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17832">issues.joomla.org/tracker/joomla-cms/17832</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by ghost** [comment]
@roland-d can you please resolve conflicting Files?

**Comment by roland-d** [comment]
@franz-wohlkoenig Not sure if it is useful to resolve merge conflicts if it is not going to be merged. Once someone is ready to merge this, let me know and I can resolve the conflicts once and for all.

**Comment by ghost** [comment]
@roland-d you're right. I was looking at RTC having conflicts hoping they get easier in Release if they get merged.

**Comment by roland-d** [comment]
@franz-wohlkoenig I found a few minutes to merge the 4.0-dev branch.

**Comment by roland-d** [comment]
@wilsonge What is this PR waiting for? Keep getting conflicts :P

**Comment by laoneo** [comment]
Thanks for the patience!

**Comment by roland-d** [comment]
@laoneo Woohoo 🎉

**Pull request merged**